### DISCUSSION TITLE
Port for ESP8266

### DISCUSSION THREAD

**Discussion by McFly-Muc** [ldijkman/randomnerd_esp32_wifi_manager#4] [Q&A]
I tried to port your sketch to the ESP8266 platform.
Changed the necessary libraries and most important switched to LittleFS instead of SPIFFS...

My only problems are these:

1. Setting variable ledPin

  // Load values saved in SPIFFS
  ssid = readFile(LittleFS, ssidPath);
  Serial.println(ssid);
  pass = readFile(LittleFS, passPath);
  Serial.println(pass);
  ip = readFile(LittleFS, ipPath);
  Serial.println(ip);
  gateway = readFile(LittleFS, gatewayPath);
  Serial.println(gateway);
  subnet = readFile(LittleFS, subnetPath);
  Serial.println(subnet);
  mdnsdotlocalurl = readFile(LittleFS, mdnsPath);
  Serial.println(mdnsdotlocalurl);
  dhcpcheck = readFile(LittleFS, dhcpcheckPath);
  Serial.println(dhcpcheck);
  relaispin = readFile(LittleFS, relaispinPath);
  Serial.println(relaispin);

  ledPin = relaispin.toInt();    <-- I get an error because ledPin was declared as a constant?  
  Serial.println(relaispin);

2. LittleFS functions

 server.on("/list", HTTP_GET, [](AsyncWebServerRequest * request) {    // /list files in spiffs on webpage
      if (!LittleFS.begin()) {
        Serial.println("An Error has occurred while mounting SPIFFS");
        return;
      }

      File root = LittleFS.open("/", "r");
      File file = root.openNextFile();
      String str = "";
      while (file) {
        str += " / ";
        str += file.name();
        str += "\r\n";
        file = root.openNextFile();
      }
      str += "\r\n";
      str += "\r\n";
      str += "totalBytes   ";  <-- Function totalBytes gives 
      str += LittleFS.totalBytes();
      str += "\r\n";
      str += "usedBytes    ";
      str += LittleFS.usedBytes();
      str += "\r\n";
      str += "freeBytes??? ";
      str += LittleFS.totalBytes()-LittleFS.usedBytes();
      str += "\r\n";
      request->send(200, "text/txt", str);
    });

I get the following error: exit status 1
'class fs::FS' has no member named 'totalBytes'

Same goes for usedBytes

Any suggestions?

**Comment by ldijkman** [comment]
Hello

Thats nice if it would run on esp8266 esp8285
Could would you maybe post the complete ino
So that i can play with it

And litlefs might be better because spiffs is  not adviced to be used anymore

Thank you for your input
Luberth

**Reply by ldijkman** [reply]
Or fork Electra to your github and do edits there

**Reply by McFly-Muc** [reply]
See attached file.

I tried to comment the two section... then it did compile but I cannot connect to my router... tried with and without DHCP...
And if the device cannot connect it retries instead of giving me the AP again...

[ESP8266_Electra_WiFi_Manager.txt](https://github.com/ldijkman/randomnerd_esp32_wifi_manager/files/7910112/ESP8266_Electra_WiFi_Manager.txt)

**Reply by ldijkman** [reply]
do you connect wifi  => direct to esp32  AccesPoint
and then  browse to 192.168.4.1

i see some hardcoded try for ip settings in code

**Reply by ldijkman** [reply]
missing parts in the code i think

maybe better to make a fork on my page to your github
and edit in github
so that i can download a complete zip

**Reply by McFly-Muc** [reply]
Yes I connect to the AP and browse to 192.168.4.1
What you see hardcoded is commented... these were the original lines from Rui...
When I look in the serial mon it tries to connect with the correct credentials but then I get the following output, this time without DHCP:

04:37:17.641 ->  ets Jan  8 2013,rst cause:2, boot mode:(3,7)
04:37:17.641 -> 
04:37:17.641 -> load 0x4010f000, len 3460, room 16 
04:37:17.679 -> tail 4
04:37:17.679 -> chksum 0xcc
04:37:17.679 -> load 0x3fff20b8, len 40, room 4 
04:37:17.679 -> tail 4
04:37:17.679 -> chksum 0xc9
04:37:17.679 -> csum 0xc9
04:37:17.679 -> v000712a0
04:37:17.679 -> ~ld
04:37:17.753 -> LittleFS mounted successfully
04:37:17.753 -> Reading file: /ssid.txt
04:37:18.763 -> xxxxxxxxxxxxx
04:37:18.763 -> Reading file: /pass.txt
04:37:19.774 -> xxxxxxxxxxxxx
04:37:19.774 -> Reading file: /ip.txt
04:37:20.798 -> 192.168.178.200
04:37:20.798 -> Reading file: /gateway.txt
04:37:21.810 -> 192.168.178.1
04:37:21.810 -> Reading file: /subnet.txt
04:37:22.787 -> 255.255.255.0
04:37:22.787 -> Reading file: /mdns.txt
04:37:23.805 -> als
04:37:23.805 -> Reading file: /dhcpcheck.txt
04:37:24.820 -> off
04:37:24.820 -> Reading file: /relaispin.txt
04:37:25.838 -> 5
04:37:25.838 -> 5
04:37:25.838 -> Reading file: /statusledpin.txt
04:37:26.854 -> 8
04:37:26.854 -> Reading file: /buttonpin.txt
04:37:27.871 -> 3
04:37:27.871 -> Reading file: /ntptime.txt
04:37:28.851 -> time.google.com
04:37:28.851 -> Reading file: /ntptimeoffset.txt
04:37:29.884 -> 0
04:37:29.884 -> Reading file: /offdelay.txt
04:37:29.884 -> - failed to open file for reading
04:37:29.884 -> 
04:37:29.884 -> h
04:37:29.884 -> Connecting to WiFi...
04:37:32.142 -> 
04:37:32.142 -> --------------- CUT HERE FOR EXCEPTION DECODER ---------------
04:37:32.142 -> 
04:37:32.142 -> Soft WDT reset
04:37:32.142 -> 
04:37:32.142 -> >>>stack>>>
04:37:32.142 -> 
04:37:32.142 -> ctx: cont
04:37:32.142 -> sp: 3ffffd00 end: 3fffffc0 offset: 01a0
04:37:32.142 -> 3ffffea0:  00dcb8cf 4bc6a7f0 3ffef7c4 3ffef67c  
04:37:32.142 -> 3ffffeb0:  00dcb8cf 00000000 00418937 00000000  
04:37:32.142 -> 3ffffec0:  4025642a 3ffef898 40100562 00000000  
04:37:32.142 -> 3ffffed0:  00dcb8cf 00000015 3ffef898 40215f3c  
04:37:32.142 -> 3ffffee0:  3fff0344 402078e0 3fffff40 3ffef67c  
04:37:32.179 -> 3ffffef0:  3ffef7c4 00000000 3ffef898 40206032  
04:37:32.179 -> 3fffff00:  40222320 00000000 3ffef898 40222320  
04:37:32.179 -> 3fffff10:  00000000 3ffe92f4 40222320 00ffffff  
04:37:32.179 -> 3fffff20:  40222320 01b2a8c0 40222320 c8b2a8c0  
04:37:32.179 -> 3fffff30:  3ffef680 00000000 00000002 40101180  
04:37:32.179 -> 3fffff40:  00000000 3ffef804 00000000 3ffef6d8  
04:37:32.179 -> 3fffff50:  3ffef6a8 3ffef804 3ffef898 40206dea  
04:37:32.179 -> 3fffff60:  00000068 3ffef6c0 00000000 feefeffe  
04:37:32.216 -> 3fffff70:  feefeffe feefeffe feefeffe feefeffe  
04:37:32.216 -> 3fffff80:  00000000 feefeffe feefeffe feefeffe  
04:37:32.216 -> 3fffff90:  feefeffe feefeffe feefeffe 3ffefa60  
04:37:32.216 -> 3fffffa0:  3fffdad0 00000000 3ffefa4c 40217f40  
04:37:32.216 -> 3fffffb0:  feefeffe feefeffe 3ffe860c 40100fd1  
04:37:32.216 -> <<<stack<<<
04:37:32.216 -> 
04:37:32.216 -> --------------- CUT HERE FOR EXCEPTION DECODER ---------------
04:37:32.254 ->

**Comment by ldijkman** [comment]
please make a fork from my page and do edits in your github
so that i can download a comlete zip
righttop corner fork button https://github.com/ldijkman/randomnerd_esp32_wifi_manager
then you have a copy on your github
edit it there

im using linux in the past i have had some strange dfferences between linux and windows arduino ide

would be better if the includ header used ifdef
so that the file can be used for both esp32 and esp8266/esp8285

example from espasycncwebserver

#include <ArduinoOTA.h>
#ifdef ESP32            ///////////////////
#include <FS.h>
#include <SPIFFS.h>
#include <ESPmDNS.h>
#include <WiFi.h>
#include <AsyncTCP.h>
#elif defined(ESP8266)   ///////////////////////////
#include <ESP8266WiFi.h>
#include <ESPAsyncTCP.h>
#include <ESP8266mDNS.h>
#endif                                       //////////////////////

#include <ESPAsyncWebServer.h>
#include <SPIFFSEditor.h>

**Reply by ldijkman** [reply]
04:37:20.798 -> 192.168.178.200
04:37:20.798 -> Reading file: /gateway.txt
04:37:21.810 -> 192.168.178.1
04:37:21.810 -> Reading file: /subnet.txt
04:37:22.787 -> 255.255.255.0

thats a ziggo modem  dutch gateway ip setting ?????????
could you use dhcp?

**Reply by McFly-Muc** [reply]
No, because the device couldn't connect to my router there was no IP assigned via DHCP... this was why I tried it with my Subnet 192.168.178.xxx

**Reply by ldijkman** [reply]
is 192.168.178.1 your gateway
lookup gateway and subnet in your wifirouter admin config page?

Me no NetWork Expert!
GateWay
Maybe IP of WiFi router Admin / Config page?
windows->DOS terminal CMD?->   ipconfig
linux->netstat -r

windos cmd ipconfig lookup gateway and subnet  
do no use given ipv4adress for  esp or your computer may conflict with esp
![image](https://user-images.githubusercontent.com/45427770/150464340-8ce16628-ffe2-48f2-92d9-2879cd17a0ba.png)

**Reply by McFly-Muc** [reply]
Yes... these values are correct...
my Gateway is 192.168.178.1

As I said, with Ruis original code (without mDNS) it works perfectly...

**Comment by ldijkman** [comment]
struct FSInfo {
    size_t totalBytes;
    size_t usedBytes;
    size_t blockSize;
    size_t pageSize;
    size_t maxOpenFiles;
    size_t maxPathLength;
};


/* https://www.mischianti.org/2020/06/22/wemos-d1-mini-esp8266-integrated-littlefs-filesystem-part-5/
 *  WeMos D1 mini (esp8266)
 *  LittleFS get info, read dir and show all file uploaded
 *  add a data folder to use with esp8266 data uploader
 *  by Mischianti Renzo <https://www.mischianti.org>
 *
 *  https://www.mischianti.org/
 *
 */
 
#include "Arduino.h"
#include "LittleFS.h"
 
void setup()
{
    Serial.begin(112500);
 
    delay(500);
 
    Serial.println(F("Inizializing FS..."));
    if (LittleFS.begin()){
        Serial.println(F("done."));
    }else{
        Serial.println(F("fail."));
    }
 
    // To format all space in LittleFS
    // LittleFS.format()
 
    // Get all information of your LittleFS
    FSInfo fs_info;
    LittleFS.info(fs_info);
 
    Serial.println("File sistem info.");
 
    Serial.print("Total space:      ");
    Serial.print(fs_info.totalBytes);
    Serial.println("byte");
 
    Serial.print("Total space used: ");
    Serial.print(fs_info.usedBytes);
    Serial.println("byte");
 
    Serial.print("Block size:       ");
    Serial.print(fs_info.blockSize);
    Serial.println("byte");
 
    Serial.print("Page size:        ");
    Serial.print(fs_info.totalBytes);
    Serial.println("byte");
 
    Serial.print("Max open files:   ");
    Serial.println(fs_info.maxOpenFiles);
 
    Serial.print("Max path length:  ");
    Serial.println(fs_info.maxPathLength);
 
    Serial.println();
 
    // Open dir folder
    Dir dir = LittleFS.openDir("/");
    // Cycle all the content
    while (dir.next()) {
        // get filename
        Serial.print(dir.fileName());
        Serial.print(" - ");
        // If element have a size display It else write 0
        if(dir.fileSize()) {
            File f = dir.openFile("r");
            Serial.println(f.size());
            f.close();
        }else{
            Serial.println("0");
        }
    }
}
 
void loop()
{
 
}
The result of the inquity


Inizializing FS...
done.
File sistem info.
Total space:      2072576byte
Total space used: 73728byte
Block size:       8192byte
Page size:        2072576byte
Max open files:   5
Max path length:  32
 
logo.jpg - 46136
provaTXT.txt - 4416

https://www.mischianti.org/2020/06/22/wemos-d1-mini-esp8266-integrated-littlefs-filesystem-part-5/

**Reply by ldijkman** [reply]
tested  it on an esp8266 4mb
same problem as you
made some changes
now i got it working

https://github.com/ldijkman/randomnerd_esp32_wifi_manager/blob/main/ESP8266-LittleFS-wifi-manager.ino

my settings for 4mb esp8266 arduino ide


https://github.com/ldijkman/randomnerd_esp32_wifi_manager/blob/main/images/20220121_172823.jpg

**Reply by McFly-Muc** [reply]
Will try it i a few hours and tell you if it works...

**Reply by McFly-Muc** [reply]
Many tanks! Just works perfectly now...  Do you know what went wrong that I couldn't connect to my router yesterday? Did you fix any other parts other than the two problems mentioned?

**Reply by ldijkman** [reply]
Serial terminal stopped at connect to station part
Crash reboot (soft wdt reset?)
Also on my side when i tested it == crash

In the wifi connect to STA part 
There whas a mili check for if not connect in certain time interval go back to AP mode

The esp8266 did not do/like it as esp32 
Now it is  an i++ count with 1 sec delay when i>=20 then do  AccessPoint mode

In loop i had to add 
MDNS.update(); // for esp8266
Otherwise i do not see its URL  in list On android bonjourbrowser app

So my example is overkill

You can do mDNS in +/-3 lines

But thanks for trying my code
